feat(recordings): Add filter action handler for overview panel

ordehi opened the pull request:
## Problem

This PR aims to reduce friction for applying filters, and leverage the existing "Pinned properties" feature so users can quickly filter for their custom properties from the overview tab.

## Changes

**Filter buttons in overview panel sidebar**
- One-click filtering from property rows
- Automatic property type detection (Event/Person/Session)
- Toast notification with "View filters" link

**Why this matters for users:**
- **Faster workflows**: Apply filters in 1 click instead of manual reconstruction
- **Better discoverability**: Filter actions appear contextually where data is shown

### Screenshots

Filter buttons in overview panel:

<img width="682" height="1012" alt="CleanShot 2025-11-11 at 18 40 58@2x" src="https://github.com/user-attachments/assets/ed99efc6-75f4-40bb-b7c1-67a42c6145a5" />

Toast notification with filters link:

<img width="884" height="194" alt="CleanShot 2025-11-11 at 18 41 35@2x" src="https://github.com/user-attachments/assets/be68b3af-f1f3-45e1-8fa1-73a424e0f3e1" />

## Testing

- Unit tests for property type classification and filter application
- Component tests for UI interactions
- Manual testing across different screen sizes and filter scenarios

Changelog: Yes - feature is complete and ready for use.

ordehi committed:
feat(recordings): Add filter action handler for overview panel

Add handleFilterByProperty function that determines property type
(person vs event) and applies filters via setFilters action.

ordehi committed:
feat(recordings): Add setFilters action access to overview panel

Import and access setFilters action from sessionRecordingsPlaylistLogic
for use in filter button handlers.

ordehi committed:
feat(recordings): Add filter button to OverviewGridItem

Add optional filter button that appears on hover with showFilter and
onFilterClick props. Button stops event propagation and has opacity
transition.

ordehi committed:
feat(recordings): Wire filter button in overview panel

Connect filter button to handleFilterByProperty for property items
with values. Empty string values are treated as filterable.

github-actions[bot] committed:
test(storybook): update UI snapshots

ordehi committed:
feat(recordings): Make overview panel filter buttons always visible

Changed filter button visibility from hover-only to always visible.
Buttons now show in secondary color by default and change to primary on hover.

ordehi committed:
fix(recordings): Fix overview panel filter buttons to use filter_group structure

Changed handleFilterByProperty to properly add filters to filter_group
instead of using deprecated person_properties/session_properties format.
This prevents malformed filter state that was causing multiple Add filter
buttons to appear.

Also fixed session properties to use PropertyFilterType.Session instead of
PropertyFilterType.Event for correct filtering.

ordehi committed:
feat(recordings): Add view filters button to toast notification

Add 'View filters' button to toast when applying filters from overview panel.
Clicking the button opens the filters panel to show the applied filters.

ordehi committed:
refactor(recordings): Make PlayerSidebarOverviewGrid reusable with optional props

Allow passing logicPropsOverride to use the component with any recording ID.
Defaults to player context if not provided, enabling reuse in tooltips and
other contexts outside the player sidebar.

ordehi committed:
refactor(recordings): Extract applyRecordingPropertyFilter to shared utils

Move filter application logic from PlayerSidebarOverviewGrid to shared utils.ts
so it can be reused in the recording preview tooltip.

ordehi committed:
feat(recordings): Add expandable preview cards with property actions

Added expandable recording preview cards that allow users to view detailed
person and property information without opening the full player. Users can
copy values or apply them as filters directly from the expanded view.

Changes:
- Moved filter application logic to shared applyRecordingPropertyFilter utility
- Added expandable details section to recording preview cards with chevron toggle
- Property details now show copy and filter action buttons
- Fixed property type classification for recordings (Event vs Person vs Session)
- Chevron stays in same position and rotates when expanding/collapsing
- Updated tests to use new shared utility and filter_group structure

ordehi committed:
Delete common/storybook/.claude/settings.local.json

github-actions[bot] committed:
test(storybook): update UI snapshots

webjunkie committed:
Merge branch 'master' into recording-overview-filter-actions

webjunkie commented:
> Got stuck trying to get storybook to update screens, not sure why that test is failing as it doesn't seem related to my changes. Welcome any tips/feedback.

Probably a problem we had in master. I merged master to try to fix.

pauldambra commented on frontend/src/scenes/session-recordings/playlist/SessionRecordingPreview.tsx:
```diff
@@ -338,7 +362,76 @@ export const SessionRecordingPreview = memo(
                         </div>
 
-                        <FirstURL startUrl={recording.start_url} />
+                        <div className="flex items-center justify-between gap-x-0.5">
```
i'm torn on this one... every button we add is one more thing for folk to understand... I actually really like the idea but haven't heard users asking for it - is it you scratching your own itch or reacting to customer demand?

could we split this into two PRs?
one to add the 'apply filter from overview'
one to add the 'show more data in the playlist'

that way we can merge or not at a different time for each

ordehi committed:
refactor(recordings): Move filter logic to sessionRecordingsPlaylistLogic action

Addresses Paul's feedback to move applyRecordingPropertyFilter from utils
to a proper action in sessionRecordingsPlaylistLogic. Also removes expandable
playlist preview feature to split into separate PR.

Changes:
- Move applyRecordingPropertyFilter logic to applyPropertyFilter action
- Update PlayerSidebarOverviewGrid to use new action
- Remove expandable preview UI from SessionRecordingPreview
- Remove util function and associated test file
- Clean up unused imports

Split strategy:
- PR 1 (this): Filter application from overview tab only
- PR 2 (next): Expandable playlist previews with filter buttons

ordehi committed:
fix: Add back UniversalFiltersGroup import in utils.ts

github-actions[bot] committed:
test(storybook): update UI snapshots

ordehi committed:
feat: Add filter tooltips and disable filtering for missing values

- Add tooltip to filter icon: "Filter for recordings matching this"
- Disable filtering when value is null or "-" with disabled state tooltip
- Visual feedback with muted styling and cursor-not-allowed for disabled filters
- Separate tooltips for value and filter icon to prevent overlap

ordehi committed:
Merge branch 'master' into recording-overview-filter-actions

github-actions[bot] committed:
test(storybook): update UI snapshots

ordehi committed:
feat: Add toggle filter functionality for recording overview properties

Clicking a filter button now toggles between three states:
- Add filter if none exists for that property
- Replace filter if same property with different value exists
- Remove filter if exact match exists

Tooltip indicates the action (Add/Replace/Remove).

ordehi committed:
feat: Improve filter button styling and layout

- Remove inconsistent icon sizing across different properties
- Add visual active state with background highlight for applied filters
- Pin filter button to the right side to prevent layout shifts
- Truncate long property values with ellipsis to prevent horizontal scrolling
- Full value still visible on hover via tooltip

ordehi committed:
fix: Prevent horizontal scrolling in overview user ID display

- Truncate long person IDs with ellipsis instead of horizontal scroll
- Keep copy button and resolution info visible
- Apply proper flex constraints to allow truncation

github-actions[bot] committed:
test(storybook): update UI snapshots

ordehi committed:
fix: TypeScript errors in filter state detection

Add explicit type guards to properly narrow filter types and
ensure operator field exists before accessing it

pauldambra reviewed (approved):
LGTM

ordehi committed:
Merge branch 'master' into recording-overview-filter-actions

ordehi committed:
chore: Merge master into recording-overview-filter-actions

  Resolved snapshot conflicts using master's version
